src/Monolog/Formatter/JsonFormatter.php

```diff
@@ -135,8 +135,12 @@ protected function formatBatchNewlines(array $records)
      *
      * @return mixed
      */
-    protected function normalize($data)
+    protected function normalize($data, $depth = 0)
```

Comment:
@Seldaek I understand the intention behind this PR reasoning behind although I am not 100% I agree with this.

Regardless what ever point of view anyone might isn't this  a BC?

```
Warning: Declaration of MyClassFormatter::normalize($data) should be compatible with Monolog\Formatter\NormalizerFormatter::normalize($data, $depth = 0)                                                                              
```

Maybe adding it in [changelog](https://github.com/Seldaek/monolog/blob/master/CHANGELOG.md#1240-2018-11-05) with a **BC* suffix?

Reply:
Well.. if it is just a warning IMO that's not really a BC Break. Do you have your own normalizer extending this? It's an easy enough fix and IMO a rare enough occurrence..

Reply:
That isn't to say it's not inconvenient.. Sorry I didn't notice this would cause a warning, but at least it's not strictly breaking anything.

Reply:
Yea.. ok.. it is an easy fix.. that's why I suggested it at least add it on readme with instructions :)

Reply:
Yup, added a notice now at https://github.com/Seldaek/monolog/releases/tag/1.24.0

Reply:
Just wanted to point out that these things could be mentioned to readme file. I don't want to appear ungrateful. You are doing a great job.

Thank you anyway.